windows/hotkey_manager_plugin.cpp

```diff
@@ -1,526 +1,477 @@
-    public:
-        static void RegisterWithRegistrar(flutter::PluginRegistrarWindows* registrar);
-
-        HotkeyManagerPlugin(flutter::PluginRegistrarWindows* registrar);
-
-        virtual ~HotkeyManagerPlugin();
-
-    private:
-        flutter::PluginRegistrarWindows* registrar;
-
-        // The ID of the WindowProc delegate registration.
-        int window_proc_id = -1;
-        
-        // Called for top-level WindowProc delegation.
-        std::optional<LRESULT> HotkeyManagerPlugin::HandleWindowProc(HWND hwnd, UINT message,
-            WPARAM wparam, LPARAM lparam);
-        void HotkeyManagerPlugin::Register(
-            const flutter::MethodCall<flutter::EncodableValue>& method_call,
-            std::unique_ptr<flutter::MethodResult<flutter::EncodableValue>> result);
-        void HotkeyManagerPlugin::Unregister(
-            const flutter::MethodCall<flutter::EncodableValue>& method_call,
-            std::unique_ptr<flutter::MethodResult<flutter::EncodableValue>> result);
-        void HotkeyManagerPlugin::UnregisterAll(
-            const flutter::MethodCall<flutter::EncodableValue>& method_call,
-            std::unique_ptr<flutter::MethodResult<flutter::EncodableValue>> result);
-
-        // Called when a method is called on this plugin's channel from Dart.
-        void HandleMethodCall(
-            const flutter::MethodCall<flutter::EncodableValue>& method_call,
-            std::unique_ptr<flutter::MethodResult<flutter::EncodableValue>> result);
-    };
-
-    // static
-    void HotkeyManagerPlugin::RegisterWithRegistrar(
-        flutter::PluginRegistrarWindows* registrar) {
-        channel =
-            std::make_unique<flutter::MethodChannel<flutter::EncodableValue>>(
-                registrar->messenger(), "hotkey_manager",
-                &flutter::StandardMethodCodec::GetInstance());
-
-        auto plugin = std::make_unique<HotkeyManagerPlugin>(registrar);
-
-        channel->SetMethodCallHandler(
-            [plugin_pointer = plugin.get()](const auto& call, auto result) {
-            plugin_pointer->HandleMethodCall(call, std::move(result));
-        });
-
-        registrar->AddPlugin(std::move(plugin));
-    }
-
-    HotkeyManagerPlugin::HotkeyManagerPlugin(flutter::PluginRegistrarWindows* registrar)
-        : registrar(registrar) {
-        window_proc_id = registrar->RegisterTopLevelWindowProcDelegate(
-            [this](HWND hwnd, UINT message, WPARAM wparam, LPARAM lparam) {
-                return HandleWindowProc(hwnd, message, wparam, lparam);
-            });
-    }
-
-    HotkeyManagerPlugin::~HotkeyManagerPlugin() {
-        registrar->UnregisterTopLevelWindowProcDelegate(window_proc_id);
-    }
-
-    void StartHandleKeyEvent() {
-        if (isStartHandleKeyEvent) return;
-        
-        isStartHandleKeyEvent = true;
-
-        MSG msg = { 0 };
-        BOOL bRet;
-
-        while ((bRet = GetMessage(&msg, NULL, 0, 0)) != 0) {
```

Comment:
Instead of running separate `GetMessage` loop, we get `wparam` from `RegisterTopLevelWindowProcDelegate` itself & pass its `HWND` when registering/unregistering hotkeys.

**(Done by this PR)**